xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-wikimacro/xwiki-platform-rendering-wikimacro-api/src/main/java/org/xwiki/rendering/macro/wikibridge/WikiMacroDescriptor.java

```diff
@@ -84,12 +85,13 @@ public Builder description(String description)
         }
 
         /**
-         * @param defaultCategory the macro default category
-         * @return this builder
+         * @param defaultCategories the macro default categories
+         * @return this build
+         * @since 14.6RC1
          */
-        public Builder defaultCategory(String defaultCategory)
+        public Builder defaultCategories(Set<String> defaultCategories)
```

Comment:
Note that you return whatever Set was provided in input, and it might be a modifiable Set (for example it's the case with your implementation of wiki macros builder) which is not great for #getDefaultCategories.

Reply:
Do you suggest to wrap the Set passed in parameter into an immutable Set internally?

Reply:
Ideally you should copy the input in a new immutable Set to be sure that there can not be any leak.

Reply:
Depends of what we use, say we choose to use `java.util.Collections#unmodifiableSet`, a Set used in parameter so we can't take any collection.

Reply:
java.util.Collections#unmodifiableSet does not created a new Set, it just wraps it.